File: backendApp/src/main/java/com/cinemas/dto/response/CelebResponse.java
package com.cinemas.dto.response;

import com.cinemas.entities.Country;
import com.cinemas.entities.Movie;
import com.cinemas.enums.RoleCeleb;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CelebResponse {
    private Integer id;

    private String name;

    private LocalDate dateOfBirth;

    private String slug;

    private String biography;

    private String description;

    private RoleCeleb role;

    private String image;

    private Country country;

    private List<MovieCelebResponse> movieList;
}

File: fllmbooking_app/lib/data/models/CelebResponse.dart
import 'package:fllmbooking_app/data/models/movie_model.dart';

class CelebResponse {
  int? id;
  String? name;
  String? description;
  DateTime? dateOfBirth;
  String? biography;
  String? role;
  String? image;
  int? countryId;
  String? slug;
  List<Movie>? movieList;

  CelebResponse({
    required this.id,
    required this.name,
    required this.description,
    required this.dateOfBirth,
    required this.biography,
    required this.role,
    required this.image,
    required this.countryId,
    required this.slug,
    required this.movieList,
  });

  factory CelebResponse.fromJson(Map<String, dynamic> json) {
    return CelebResponse(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      dateOfBirth: json['dateOfBirth'] != null
          ? DateTime.parse(json['dateOfBirth'])
          : null,
      biography: json['biography'],
      role: json['role'],
      image: json['image'],
      countryId: json['countryId'],
      slug: json['slug'],
      movieList: json['movieList'] != null
          ? (json['movieList'] as List)
              .map((movie) => Movie.fromJson(movie))
              .toList()
          : null,
    );
  }
}
